### PR TITLE
MH-13074, Clean up asset manager REST endpoints

### PR DESCRIPTION
**Pull request by lkiesow**
The asset manager REST endpoints contains a few pieces of weird or
duplicated code:

- Remove Entwine's functional library for exception handling
- Do not duplicate identical methods
- Remove broken “If-None-Match” header parameter

### CODE REVIEW

File: modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java [hunk 1]
```diff
@@ -115,12 +111,7 @@
       returnDescription = "No content is returned.")
   @Deprecated
   public Response add(@FormParam("mediapackage") final MediaPackageImpl mediaPackage) {
-    return handleException(new P1Lazy<Response>() {
-      @Override public Response get1() {
-        getAssetManager().takeSnapshot(DEFAULT_OWNER, mediaPackage);
-        return noContent();
-      }
-    });
+    return snapshot(mediaPackage);
```
Comment (lkiesow) [inline modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java:114]: This is identical to `snapshot` anyway

File: modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java [hunk 2]
```diff
@@ -272,51 +261,42 @@ public Response getMediaPackage(@PathParam("mediaPackageID") final String mediaP
               responseCode = SC_INTERNAL_SERVER_ERROR)})
   public Response getAsset(@PathParam("mediaPackageID") final String mediaPackageID,
                            @PathParam("mediaPackageElementID") final String mediaPackageElementID,
-                           @PathParam("version") final String version,
-                           @HeaderParam("If-None-Match") final String ifNoneMatch) {
-    return handleException(new P1Lazy<Response>() {
-      @Override public Response get1() {
-        if (StringUtils.isNotBlank(ifNoneMatch)) {
```
Comment (lkiesow) [inline modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java:279]: This is undocumented and makes no sense at all

File: modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java [hunk 3]
```diff
@@ -272,51 +261,42 @@ public Response getMediaPackage(@PathParam("mediaPackageID") final String mediaP
...
-      Throwable cause = e;
-      if (e instanceof RuntimeException && e.getCause() != null) {
-        cause = ((RuntimeException)e).getCause();
-      }
-      if (cause instanceof UnauthorizedException) {
-        return forbidden();
-      }
-
-      throw new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR);
+  public static Response handleException(Exception e) {
```
Comment (lkiesow) [inline modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java:290]: This non-functional programming approach is actually shorter, arguably easier to read and does not require Entwine's functional library

File: modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java [hunk 4]
```diff
@@ -179,19 +170,18 @@ public Response snapshot(@FormParam("mediapackage") final MediaPackageImpl media
...
-          final AQueryBuilder q = getAssetManager().createQuery();
-          if (q.delete(AssetManager.DEFAULT_OWNER, q.snapshot()).where(q.mediaPackageId(mediaPackageId)).run() > 0) {
-            return noContent();
-          } else {
-            return notFound();
-          }
-        } else
-          return notFound();
+    if (StringUtils.isEmpty(mediaPackageId)) {
+      return notFound();
```
Comment (gregorydlogan) [inline modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java:174]: Ugh, shouldn't this be badRequest()?  Nevermind, it's the same (incorrect) behaviour.

**Pull request merged**